x/confidentialtransfers/types/zk.go

```diff
@@ -419,3 +419,95 @@ func (a *Auditor) FromProto() (*TransferAuditor, error) {
 		TransferAmountHiEqualityProof: transferAmountHiEqualityProof,
 	}, nil
 }
+
+func (z *ZeroBalanceProof) Validate() error {
+	if z.YP == nil || z.YD == nil || z.Z == nil {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "zero proof is invalid")
+	}
+	return nil
+}
+
+func (z *ZeroBalanceProof) ToProto(zkp *zkproofs.ZeroBalanceProof) *ZeroBalanceProof {
+	return &ZeroBalanceProof{
+		YP: zkp.Yp.ToAffineCompressed(),
+		YD: zkp.Yd.ToAffineCompressed(),
+		Z:  zkp.Z.Bytes(),
+	}
+}
+
+func (z *ZeroBalanceProof) FromProto() (*zkproofs.ZeroBalanceProof, error) {
+	err := z.Validate()
+	if err != nil {
+		return nil, err
+	}
+	ed25519Curve := curves.ED25519()
+
+	yp, err := ed25519Curve.Point.FromAffineCompressed(z.YP)
+	if err != nil {
+		return nil, err
+	}
+
+	yd, err := ed25519Curve.Point.FromAffineCompressed(z.YD)
+	if err != nil {
+		return nil, err
+	}
+
+	zScalar, err := ed25519Curve.Scalar.SetBytes(z.Z)
+	if err != nil {
+		return nil, err
+	}
+
+	return &zkproofs.ZeroBalanceProof{
+		Yp: yp,
+		Yd: yd,
+		Z:  zScalar,
+	}, nil
+}
+
+type CloseAccountProofs struct {
+	ZeroAvailableBalanceProof *zkproofs.ZeroBalanceProof
+	ZeroPendingBalanceLoProof *zkproofs.ZeroBalanceProof
+	ZeroPendingBalanceHiProof *zkproofs.ZeroBalanceProof
+}
+
+func (c *CloseAccountProof) Validate() error {
+	if c.ZeroAvailableBalanceProof == nil || c.ZeroPendingBalanceLoProof == nil || c.ZeroPendingBalanceHiProof == nil {
```

Comment:
Just a thought: Should we make the validate() calls chain validate the child proofs here as well (instead of just checking for nil). Seems like it would be what the validate call should do instead of making that the responsibility of FromProto.